src/amo/components/SearchForm.js

```diff
@@ -25,8 +25,9 @@ export class SearchFormBase extends React.Component {
 
   goToSearch(query) {
     const { api, pathname } = this.props;
+    const encodedQuery = query ? encodeURIComponent(query) : '';
     this.context.router.push(
```

Comment:
I think it would be safer to rely on the [router API](https://github.com/ReactTraining/react-router/blob/master/docs/API.md#pushpathorloc):
````js
this.context.router.push({ pathname: ..., query: ... });
````